### PR TITLE
model functions

### PR DESCRIPTION
**Pull request by Unnunu**

### CODE REVIEW

**Comment by BowserSlug** [comment]
🚀🚀🚀🚀 This PR matches 4 functions (+0.28%) on `us`.

File: include/model.h [hunk 1]
```diff
@@ -142,6 +142,14 @@ typedef enum ShapeTypes {
     SHAPE_TYPE_SPECIAL_GROUP = 10,
 } ShapeTypes;
 
+typedef enum ExtraTileTypes {
+    EXTRA_TILE_0 = 0,
```
Comment (HailSanta) [inline include/model.h:146]: These are known and should be
EXTRA_TILE_NONE = 0,
EXTRA_TILE_MIPMAPS = 1,
EXTRA_TILE_AUX_SAME = 2, // has aux tile with same settings as main
EXTRA_TILE_AUX_INDEPENDENT = 3, // has aux tile with different settings

File: src/a5dd0_len_114e0.c [hunk 2]
```diff
@@ -2988,8 +2988,638 @@ void state_render_frontUI(void) {
...
+                    break;
+            }
+        }
+    }
+
+    if (textureHeader != NULL) {
+        switch (extraTileType) {
+            case EXTRA_TILE_3:
+            case EXTRA_TILE_4:
+                prop = get_model_property(modelNode, MODEL_PROP_KEY_SPECIAL);
```
Comment (HailSanta) [inline src/a5dd0_len_114e0.c:3126]: The two-s32 values for this property are:
00AAABBB 000CD0EF
A = auxOffsetT
B = auxOffsetS
C = auxShiftT
D = auxShiftS
E = ModelReplaceType (an enum)
F = defaultScrollUnit

File: src/a5dd0_len_114e0.c [hunk 3]
```diff
@@ -2988,8 +2988,638 @@ void state_render_frontUI(void) {
...
+        if (renderMode != RENDER_MODE_ALPHATEST && renderMode != RENDER_MODE_ALPHATEST_ONESIDED) {
+            *(*gfxPos) = D_8014B0B8[v1][combineSubType];
+        } else {
+            *(*gfxPos) = D_8014B400[v1][combineSubType];
+        }
+        (*gfxPos)++;
+    }
+
+    switch (s3) {
+        case 1:
```
Comment (HailSanta) [inline src/a5dd0_len_114e0.c:3171]: This is some 'texturing mode' enum, with at least these values:
1 = NO_TEXTURE
2 = NORMAL
3 = FOG
6 = FOG_MULTIPLY_COLOR

**Pull request merged**